src/gui/dialogs/units_dialog.hpp

```diff
@@ -0,0 +1,247 @@
+	}
+
+	units_dialog& set_ok_label(const std::string& ok_label)
+	{
+		ok_label_ = ok_label;
+		return *this;
+	}
+
+	units_dialog& show_variations(const bool show_variation)
+	{
+		show_variation_grid_ = show_variation;
+		return *this;
+	}
+
+	units_dialog& show_gender(const bool show_gender)
+	{
+		show_gender_grid_ = show_gender;
+		return *this;
+	}
+
+	units_dialog& show_dismiss_option(const bool show_dismiss)
+	{
+		show_dismiss_ = show_dismiss;
+		return *this;
+	}
+
+	units_dialog& show_rename_option(const bool show_rename)
+	{
+		show_rename_ = show_rename;
+		return *this;
+	}
+
+	units_dialog& show_all_headers()
+	{
+		visible_headers_.set();
+		return *this;
+	}
+
+	units_dialog& hide_all_headers()
+	{
+		visible_headers_.reset();
+		return *this;
+	}
+
+	units_dialog& show_header(const int col_num, const bool visible = true)
+	{
+		visible_headers_[col_num] = visible;
+		return *this;
+	}
+
+	units_dialog& set_help_topic(const std::string& topic_id)
+	{
+		topic_id_ = topic_id;
+		return *this;
+	}
+
+	units_dialog& set_team(team* team)
+	{
+		team_ = team;
+		return *this;
+	}
+
+	units_dialog& set_types(const std::vector<const unit_type*>& types)
+	{
+		unit_type_list_ = types;
+		return *this;
+	}
+
+	units_dialog& set_units(const std::vector<unit_const_ptr>& units)
+	{
+		unit_list_ = units;
+		return *this;
+	}
+
+	units_dialog& set_row_num(const size_t row_num)
+	{
+		row_num_ = row_num;
+		return *this;
+	}
+
+	/**
+	 * Corresponding to each widget in the row with id 'id', there is a lambda that generates
+	 * the corresponding label to set to that widget. This method sets those generator functions.
+	 * The 'id's must be the same as those defined for the various widgets
+	 * in the [list_definition] in the WML file.
+	 */
+	template<typename Value, typename Generator = std::function<std::string(const Value&)>>
+	units_dialog& set_column_generator(
+		const std::string& id,
+		const std::vector<Value>& container,
+		const Generator& generator,
+		const bool use_as_sorter = false)
+	{
+		column_generators_.try_emplace(id, [&container, generator](size_t index) { return generator(container[index]); });
+		// use the generator function also as sorter function
+		if (use_as_sorter) {
+			find_widget<gui2::listbox>("recall_list").register_sorting_option(
+				column_generators_.size()-1,
+				[&container, generator](size_t index) { return generator(container[index]); });
+		}
+		return *this;
+	}
+
+	/**
+	 * Sets the generator function for the tooltips
+	 */
+	units_dialog& set_tooltip_generator(const std::function<std::string(const int)>& generator)
+	{
+		tooltip_gen_ = generator;
+		return *this;
+	}
+
+	template<typename Func>
+	units_dialog& set_sorter(int col, const Func& sorter)
+	{
+		find_widget<gui2::listbox>("recall_list").register_sorting_option(col, sorter);
+		return *this;
+	}
+
+	template<typename Value, typename Generator = std::function<std::string(const Value&)>>
+	units_dialog& set_translatable_sorter(
```

Comment:
This doesn't need to be its own function now that the API has been unified. You can just rely on `set_column_generator`.

Reply:
So just `set_column_generator(...., true)` right?

Reply:
removed.